presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxSplitManager.java

```diff
@@ -57,7 +58,9 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
         TupleDomain<ColumnHandle> predicate = jmxLayout.getConstraint();
 
         //TODO is there a better way to get the node column?
-        JmxColumnHandle nodeColumnHandle = tableHandle.getColumns().get(0);
+        JmxColumnHandle nodeColumnHandle = tableHandle.getColumns().stream()
+                .filter(jmxColumnHandle -> jmxColumnHandle.getColumnName().equals(NODE_COLUMN_NAME))
+                .findFirst().get();
```

Comment:
if a MBean exposes value called "node" then yes, there can be another one :/ In JmxMetadata we add two columns at begining `NODE_COLUMN_NAME` and `TIMESTAMP_COLUMN_NAME` so we at least know that those two special columns will be at front. Maybe there should be some assertion, but I don't know - didn't want to change behavior.